app/services/exchange_rate_service.rb

```diff
@@ -0,0 +1,61 @@
+class ExchangeRateService
+  API_URL = 'https://api.exchangeratesapi.io/latest'.freeze
+  EXCHANGE_RATES_UTC_REFRESH_HOUR = 15
+  EXCHANGE_RATES_UTC_REFRESH_MIN  = 15
+  REDIS_KEY = 'trade-tariff-exchange-rates'.freeze
+
+  def call
+    cached_exchange_rates || fetched_exchange_rates
+  end
+
+  private
+
+  def cached_exchange_rates(ignore_expiry: false)
+    response = TradeTariffBackend.redis.get(REDIS_KEY)
+
+    if response.present?
+      exchange_rates = JSON.parse(response)
+      exchange_rates['expires_at'] = Time.zone.parse(exchange_rates['expires_at'])
+
+      return exchange_rates if ignore_expiry
+      return exchange_rates if exchange_rates['expires_at'] > now
+
+      false
```

Comment:
Should this return a boolean? How about nil maybe?

Reply:
Yeah, good shout